### PR TITLE
Base spider for doem.org.br and implementation for BA/Acajutiba

### LINKED ISSUE #300: Add base spider for Diário Oficial Eletrônico dos Municípios
Different cities are under http://doem.org.br/ domain. We could benefit from it having a base spider for this source.

Examples (only from Bahia):

* Acajutiba BA
* Alagoinhas BA
* Cachoeira BA
* Campo Formoso BA
* Catu BA
* Ipiaú BA
* Itapetinga BA
* Juazeiro BA
* Mucuri BA
* Porto Seguro BA
* Ribeira do Pombal BA
* Santo Amaro BA
* Senhor do Bonfim BA
* São Gonçalo dos Campos BA
* Tucano BA

### PR DESCRIPTION
**Pull request by rodbv**
Closes #300 

- Created a base spider for DOEM
- Added one spider for BA/Acajutiba to use the new base spider

Having that working, we can just add several other spiders like Acajutiba's pretty easily!

### CODE REVIEW

File: processing/data_collection/gazette/spiders/base.py [hunk 1]
```diff
@@ -186,3 +186,83 @@ def get_gazette(self, document):
...
+        return f"https://doem.org.br/{state}/{city}/pesquisar?data_inicial=2013-01-01&page={page}"
+
+    @staticmethod
+    def get_last_page(response):
+        """
+        Gets the last page number available in the pages navigation menu
+        """
+        PAGINATION_CSS_QUERY = "ul.pagination li a::text"
+        pages = response.css(PAGINATION_CSS_QUERY).getall()
+        return max([int(page) for page in pages if page.isnumeric()])
```
Comment (jvanz) [inline processing/data_collection/gazette/spiders/base.py:240]: I like small and succinct methods. Really cool! Just a curiosity, why `@staticmethod` ?

File: processing/data_collection/gazette/spiders/base.py [hunk 2]
```diff
@@ -186,3 +186,83 @@ def get_gazette(self, document):
...
+            yield Gazette(
+                date=date,
+                file_urls=[file_url],
+                edition_number=edition_number,
+                power="executive_legislative",
+            )
+
+    def get_url(self, page=1):
+        state, city = self.name.split("_")
+        return f"https://doem.org.br/{state}/{city}/pesquisar?data_inicial=2013-01-01&page={page}"
```
[the remark this answers is not in the document]
Comment (rodbv) [reply processing/data_collection/gazette/spiders/base.py:231]: Yep there are dozens of cities there, once we get the base spider approved we will be able to add quite a lot of them

File: processing/data_collection/gazette/spiders/base.py (hunk 1 above)
Comment (rodbv) [reply processing/data_collection/gazette/spiders/base.py:240]: I saw this being done on a few spiders that I checked for reference, and I assumed that was the "convention" for  these accessory methods that don't really need any attribute of the class. I'm glad to change if you think it's better to have them all as class methods.

File: processing/data_collection/gazette/spiders/base.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (rodbv) [reply processing/data_collection/gazette/spiders/base.py:231]: Sure!

File: processing/data_collection/gazette/spiders/base.py (hunk 1 above)
Comment (jvanz) [reply processing/data_collection/gazette/spiders/base.py:243]: Hmm, yes, please do it. As they are used only in this class, I think it make more sense.

**Review by jvanz — CHANGES_REQUESTED** [review]

File: processing/data_collection/gazette/spiders/base.py (hunk 1 above)
Comment (rodbv) [reply processing/data_collection/gazette/spiders/base.py]: Done!

**Comment by rodbv** [comment]
Argh not sure why black changed all these old spiders :(

**Comment by rodbv** [comment]
I've created PR #347 which a branch created with the new project structure, it will be too much trouble to rebase  this branch.

**Pull request closed without merging**